druntime/src/core/attribute.d

```diff
@@ -236,10 +236,18 @@ version (UdaGNUAbiTag) struct gnuAbiTag
 {
     string[] tags;
 
-    this(string[] tags...)
+    this(return scope string[] tags)
```

Comment:
Isn't this struct meant to be used at CTFE only?
Surely then the appropriate migration path would be with a tuple parameter? @Geod24

Reply:
To clarify, it seems to me that we can get away with just:
```suggestion
    this(T...)(T tags)
        if (allSatisy!(isStringValue, tags))
    {
        static foreach (tag; tags)
            this.tags ~= tag;
    }
```
And not bother with the extra backwards compatible constructors (that aren't really backwards compatible).

Reply:
This PR is not an attempt to redesign it, it's just to make it DIP1000 compatible. I suggest a redesign should be a separate effort.

Reply:
The signature change in this PR is itself a redesign that breaks downstream user code.

Reply:
@WalterBright Indeed, this change is a breaking change.